src/main.tsx

```diff
@@ -1,10 +1,55 @@
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import React from "react";
-import ReactDOM from "react-dom/client";
-import App from "./App";
-import "./App.css";
-
-ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>
-);
+import ReactDOM from "react-dom";
+import { createBrowserRouter, RouterProvider } from "react-router-dom";
+import { Header } from "./routes/headers";
+import { MangaCardContainer } from "./routes/mangaCard";
+import { MangaPage } from "./routes/mangaPages";
+
+/**
+ * The QueryClient instance used to make queries
+ */
+const queryClient = new QueryClient();
+
+/**
+ * The configuration object for the BrowserRouter
+ */
+const routerConfig = [
+  {
+    path: "/",
+    element: <Header />,
+    children: [
+      {
+        path: "mangalist/:orderType",
+        element: <MangaCardContainer />,
+      },
+      {
+        path: "manga/:mangaId",
+        element: <MangaPage />,
+      },
+    ],
+  },
+];
+
+/**
+ * The BrowserRouter instance
+ */
+const router = createBrowserRouter(routerConfig);
```

Comment:
Nice, routerConfig allows for modularity.

Perhaps make a separate tsx file that creates and returns your RouterProvider using this config.